### PR TITLE
add Nuri Exchange Scroll

### PR DESCRIPTION
**Pull request by 0xShad0w**

### CODE REVIEW

File: integrations/nuri.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                    nfp_manager.events.Transfer(),
+                    start_block,
+                    to_block,
+                )
+                for transfer in transfers:
+                    all_users.add(transfer["args"]["to"])
+            except Exception as e:
+                print(f"Error fetching transfers from block {start_block} to {to_block}: {e}")
+            
+            start_block = to_block + 1
```
Comment (FJ-Riveros) [inline integrations/nuri.py:111]: I believe here its
```suggestion
            start_block += page_size
```

**Review by FJ-Riveros — CHANGES_REQUESTED** [review]
Could you please take a look at the comments and also fix the merge conflicts please?

File: integrations/nuri.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            to_block = min(start_block + page_size, target_block)
+            try:
+                transfers = fetch_events_logs_with_retry(
+                    f"Nuri users from {start_block} to {to_block}",
+                    nfp_manager.events.Transfer(),
+                    start_block,
+                    to_block,
+                )
+                for transfer in transfers:
+                    all_users.add(transfer["args"]["to"])
```
[the remark this answers is not in the document]
Comment (0xShad0w) [reply integrations/nuri.py]: Thanks for the suggestion. I'm querying the pool directly now instead of the nfpManager contract. Should save some requests

**Review by FJ-Riveros — APPROVED** [review]
LGTM

**Pull request merged**